src/uniswap-summaries.md

```diff
@@ -3439,6 +3439,40 @@ module SOLIDITY-UNISWAP-MINT-SUMMARY
 endmodule
 ```
 
+```k
+module SOLIDITY-UNISWAP-APPROVE-SUMMARY
+  imports SOLIDITY-CONFIGURATION
+  imports SOLIDITY-EXPRESSION
+  imports SOLIDITY-UNISWAP-TOKENS
+
+  syntax KItem ::= "stopFunction"
+
+  rule <k> bind ( _STORE,
+                  ListItem ( usr ) ListItem ( wad ),
+                  ListItem ( address ) ListItem ( uint256 ) ,
+                  v ( V1:MInt{160} , address ),
+                  v ( V2:MInt{256} , uint256 ),
+                  .TypedVals , ListItem ( bool ) , ListItem ( noId ) ) ~> allowance [ msg . sender ] [ usr ] = wad ;  emit approvalEvent ( msg . sender , usr , wad , .TypedVals ) ;  return true ;  .Statements ~> return void ; ~> .K =>return v ( true , bool ) ; ... </k>
+       <summarize> true </summarize>
+        <this> THIS </this>
+        <contract-address> THIS </contract-address>
+        <this-type> TYPE </this-type>
+        <contract-id> TYPE </contract-id>
+        <current-function> approve </current-function>
+        <store> S => S ListItem(V1) // usr
+                       ListItem(V2) // wad
+        </store>
+        <env>
+          ENV => ENV [ usr <- var(size(S)       , address) ]
+                     [ wad <- var(size(S) +Int 1, uint256) ]
+        </env>
+        <msg-sender> SENDER </msg-sender>
+        <contract-storage>
+          Storage => Storage [ allowance <- write( {Storage [ allowance ] orDefault .Map}:>Value, ListItem(SENDER), (V1 |-> V2:MInt{256}), (mapping ( address daiownr => mapping ( address daispdr => uint256))) )]
```

Comment:
Can you explain this?

Reply:
I think I get it, following the rules for write, but because it is the first I am seeing it I want to be sure.

Reply:
Sure!
This is the 
https://github.com/Pi-Squared-Inc/solidity-demo-semantics/blob/e8ade4c8d6680aa0dfe08298d9727b7a85350215/test/examples/swaps/UniswapV2SwapRenamed.sol#L563
It's easy to understand and visualize when we break this into more lines, but I admit this was why I got stuck in this rule last Friday.
```
Storage => Storage [ allowance <- 
    write( {Storage [ allowance ] orDefault .Map}:>Value,  // This gets us the map element on the `Storage` cell 
           ListItem(SENDER),                               // This says that we want to write to `allowance[SENDER]`
           (V1 |-> V2:MInt{256}),                          // As we're not reusing the value of `allowance[SENDER][V1]`, we can just initiate a new mapping that will be at `allowance[SENDER]`, then we'll have a two dimension array with V2 at allowance[SENDER][V1] position.
           (mapping ( address daiownr => mapping ( address daispdr => uint256))) // This is the type of the structure we're writing to.
         )
]
```

Reply:
After thinking more about it, this doesn't seem correct. This is overwriting all of the contents of the map in `allowance[sender]` with the map `(V1 |-> V2)`. What if `allowance[sender]` contains the map `V1|->X V2 |-> Y`?
Then `V2 |-> Y` would be deleted.
I think that what we want here is to rewrite only the position allowance[sender][V1]. Meaning, another write should be used.
What do you think?